arbos/arbosState/arbosstate.go

```diff
@@ -313,6 +313,9 @@ func (state *ArbosState) UpgradeArbosVersion(
 			if oldAmortizationCap == math.MaxUint64 {
 				ensure(state.l1PricingState.SetAmortizedCostCapBips(0))
 			}
+
+			// Clear chainOwners list to allow rectification of the mapping.
+			ensure(state.chainOwners.ClearList())
```

Comment:
Let's only do this `if !firstTime`, because newly created chains won't have the same issue but might have an initial owner. This should let you remove the new call you have in TestArbOwner